### PR TITLE
Adding custom scope handler

### PR DESCRIPTION
**Pull request by Abilashini**

### CODE REVIEW

File: custom-scope-handler/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+## Custom Scope Handler
+
+### Introduction / Use case.
+This custom scope handler can be used to remove the scopes other than the allowed scopes from the token request. (This sample considers Scope_A, Scope_B and Scope_C as allowed scopes)
+
+### Applicable product versions.
+WSO2 IS 5.7.0
```
Comment (vihanga-liyanage) [inline custom-scope-handler/README.md]: Shall we make this line as follows?
```Tested with WSO2 IS 5.7.0```

File: custom-scope-handler/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+### How to use.
+1. Build the custom-scope-handler using the command `mvn clean install`.
+2. Copy the CustomScopeHandler-1.0.0.jar located inside the target to the directory, <IS_Home>/repository/components/dropins
```
Comment (vihanga-liyanage) [inline custom-scope-handler/README.md]: Suggestion:
Copy the CustomScopeHandler-1.0.0.jar located inside the target to the directory, **<IS_Home>/repository/components/dropins**

File: custom-scope-handler/README.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+3. Add the classpath of the scope handler to the identity.xml file located at <IS_Home>/repository/conf/identity under the tag ScopeHandlers
```
Comment (vihanga-liyanage) [inline custom-scope-handler/README.md]: Suggestion:
Add the classpath of the scope handler to the identity.xml file located at **<IS_Home>/repository/conf/identity** directory under the tag ScopeHandlers.
`<ScopeHandler class="org.wso2.custom.scope.handler.CustomScopeHandler" />`

File: custom-scope-handler/README.md (hunk 2 above)
Comment (Abilashini) [reply custom-scope-handler/README.md]: There is no difference between the suggestion and the existing line. Please verify the suggestion.

File: custom-scope-handler/src/main/java/org/wso2/custom/scope/handler/internal/CustomScopeHandlerDataHolder.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.wso2.custom.scope.handler.internal;
+
+import org.wso2.carbon.user.core.service.RealmService;
+
+public class CustomScopeHandlerDataHolder {
+    private static CustomScopeHandlerDataHolder instance = new CustomScopeHandlerDataHolder();
```
Comment (omindu) [inline custom-scope-handler/src/main/java/org/wso2/custom/scope/handler/internal/CustomScopeHandlerDataHolder.java]: Add a new line

File: custom-scope-handler/src/main/java/org/wso2/custom/scope/handler/internal/CustomScopeHandlerServiceComponent.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.wso2.carbon.user.core.service.RealmService;
+
+@Component(name = "org.wso2.custom.scope.handler", immediate = true)
+public class CustomScopeHandlerServiceComponent {
+    private static Log log = LogFactory.getLog(CustomScopeHandlerServiceComponent.class);
```
Comment (omindu) [inline custom-scope-handler/src/main/java/org/wso2/custom/scope/handler/internal/CustomScopeHandlerServiceComponent.java]: Add a new line

File: custom-scope-handler/pom.xml [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+                            org.wso2.custom.scope.handler.*; version="${project.version}",
+                        </Export-Package>
+                        <Import-Package>
+                            org.osgi.framework.*,
+                            org.osgi.service.component.*,
+                            org.apache.commons.logging.*,
+                            org.wso2.carbon.identity.oauth2.*,
+                            org.wso2.carbon.user.core.*
+                        </Import-Package>
+                        <Dynamic-Import>true</Dynamic-Import>
```
Comment (omindu) [inline custom-scope-handler/pom.xml]: Do we need dynamic imports here ?

File: custom-scope-handler/src/main/java/org/wso2/custom/scope/handler/internal/CustomScopeHandlerDataHolder.java (hunk 4 above)
Comment (Abilashini) [reply custom-scope-handler/src/main/java/org/wso2/custom/scope/handler/internal/CustomScopeHandlerDataHolder.java]: fixed at https://github.com/wso2-incubator/samples-is/pull/17

File: custom-scope-handler/src/main/java/org/wso2/custom/scope/handler/internal/CustomScopeHandlerServiceComponent.java (hunk 5 above)
Comment (Abilashini) [reply custom-scope-handler/src/main/java/org/wso2/custom/scope/handler/internal/CustomScopeHandlerServiceComponent.java]: fixed at https://github.com/wso2-incubator/samples-is/pull/17

File: custom-scope-handler/pom.xml (hunk 6 above)
Comment (Abilashini) [reply custom-scope-handler/pom.xml]: removed dynamic imports at https://github.com/wso2-incubator/samples-is/pull/17

**Pull request merged**